internal/controllers/cluster/cluster_controller.go

```diff
@@ -74,6 +80,25 @@ type Reconciler struct {
 	externalTracker external.ObjectTracker
 }
 
+// LoggerCustomizer
+// FIXME: will be moved to a util package
+func LoggerCustomizer(log logr.Logger, controllerName, kind string) func(_ logr.Logger, req reconcile.Request) logr.Logger {
+	// FIXME: just a hack for now
+	var rngSeed int64
+	_ = binary.Read(crand.Reader, binary.LittleEndian, &rngSeed)
+	randSource := rand.New(rand.NewSource(rngSeed))
+
+	return func(_ logr.Logger, req reconcile.Request) logr.Logger {
```

Comment:
Note: we can either create a new logger e.g. based on `mgr.GetLogger()` or modify the one passed in by controller-runtime here.